taken from upstream glibc ports git repo

From 65342356965f356a7fb9b4578856c5ad0861d423 Mon Sep 17 00:00:00 2001
From: Aurelien Jarno <aurelien@aurel32.net>
Date: Tue, 1 Mar 2011 10:00:28 +1000
Subject: [PATCH] alpha: Define DEFAULT_STACK_PERMS

---
 ChangeLog.alpha           |    4 ++++
 sysdeps/alpha/stackinfo.h |    6 ++++++
 2 files changed, 10 insertions(+), 0 deletions(-)

2011-02-28  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/alpha/stackinfo.h: Define DEFAULT_STACK_PERMS with PF_X.

diff --git a/sysdeps/alpha/stackinfo.h b/sysdeps/alpha/stackinfo.h
index 0a281bd..d9dbc35 100644
--- ports/sysdeps/alpha/stackinfo.h
+++ ports/sysdeps/alpha/stackinfo.h
@@ -22,7 +22,13 @@
 #ifndef _STACKINFO_H
 #define _STACKINFO_H	1
 
+#include <elf.h>
+
 /* On Alpha the stack grows down.  */
 #define _STACK_GROWS_DOWN	1
 
+/* Default to an executable stack.  PF_X can be overridden if PT_GNU_STACK is
+ * present, but it is presumed absent.  */
+#define DEFAULT_STACK_PERMS (PF_R|PF_W|PF_X)
+
 #endif	/* stackinfo.h */
-- 
1.7.4.1

